File: src/frontend/cadastros/lib/app/data/model/sindicato_model.dart
import 'dart:convert';
import 'package:pluto_grid/pluto_grid.dart';

import 'package:cadastros/app/infra/infra_imports.dart';
import 'package:intl/intl.dart';
import 'package:cadastros/app/data/domain/domain_imports.dart';

class SindicatoModel {
	int? id;
	String? nome;
	int? codigoBanco;
	int? codigoAgencia;
	String? contaBanco;
	String? codigoCedente;
	String? logradouro;
	String? numero;
	String? bairro;
	int? municipioIbge;
	String? uf;
	String? fone1;
	String? fone2;
	String? email;
	String? tipoSindicato;
	DateTime? dataBase;
	double? pisoSalarial;
	String? cnpj;
	String? classificacaoContabilConta;

	SindicatoModel({
		this.id,
		this.nome,
		this.codigoBanco,
		this.codigoAgencia,
		this.contaBanco,
		this.codigoCedente,
		this.logradouro,
		this.numero,
		this.bairro,
		this.municipioIbge,
		this.uf,
		this.fone1,
		this.fone2,
		this.email,
		this.tipoSindicato,
		this.dataBase,
		this.pisoSalarial,
		this.cnpj,
		this.classificacaoContabilConta,
	});

	static List<String> dbColumns = <String>[
		'id',
		'nome',
		'codigo_banco',
		'codigo_agencia',
		'conta_banco',
		'codigo_cedente',
		'logradouro',
		'numero',
		'bairro',
		'municipio_ibge',
		'uf',
		'fone1',
		'fone2',
		'email',
		'tipo_sindicato',
		'data_base',
		'piso_salarial',
		'cnpj',
		'classificacao_contabil_conta',
	];
	
	static List<String> aliasColumns = <String>[
		'Id',
		'Nome',
		'Codigo Banco',
		'Codigo Agencia',
		'Conta Banco',
		'Codigo Cedente',
		'Logradouro',
		'Numero',
		'Bairro',
		'Municipio Ibge',
		'Uf',
		'Fone1',
		'Fone2',
		'Email',
		'Tipo Sindicato',
		'Data Base',
		'Piso Salarial',
		'Cnpj',
		'Classificacao Contabil Conta',
	];

	SindicatoModel.fromJson(Map<String, dynamic> jsonData) {
		id = jsonData['id'];
		nome = jsonData['nome'];
		codigoBanco = jsonData['codigoBanco'];
		codigoAgencia = jsonData['codigoAgencia'];
		contaBanco = jsonData['contaBanco'];
		codigoCedente = jsonData['codigoCedente'];
		logradouro = jsonData['logradouro'];
		numero = jsonData['numero'];
		bairro = jsonData['bairro'];
		municipioIbge = jsonData['municipioIbge'];
		uf = SindicatoDomain.getUf(jsonData['uf']);
		fone1 = jsonData['fone1'];
		fone2 = jsonData['fone2'];
		email = jsonData['email'];
		tipoSindicato = SindicatoDomain.getTipoSindicato(jsonData['tipoSindicato']);
		dataBase = jsonData['dataBase'] != null ? DateTime.tryParse(jsonData['dataBase']) : null;
		pisoSalarial = jsonData['pisoSalarial']?.toDouble();
		cnpj = jsonData['cnpj'];
		classificacaoContabilConta = jsonData['classificacaoContabilConta'];
	}

	Map<String, dynamic> get toJson {
		Map<String, dynamic> jsonData = <String, dynamic>{};

		jsonData['id'] = id != 0 ? id : null;
		jsonData['nome'] = nome;
		jsonData['codigoBanco'] = codigoBanco;
		jsonData['codigoAgencia'] = codigoAgencia;
		jsonData['contaBanco'] = contaBanco;
		jsonData['codigoCedente'] = codigoCedente;
		jsonData['logradouro'] = logradouro;
		jsonData['numero'] = numero;
		jsonData['bairro'] = bairro;
		jsonData['municipioIbge'] = municipioIbge;
		jsonData['uf'] = SindicatoDomain.setUf(uf);
		jsonData['fone1'] = Util.removeMask(fone1);
		jsonData['fone2'] = Util.removeMask(fone2);
		jsonData['email'] = email;
		jsonData['tipoSindicato'] = SindicatoDomain.setTipoSindicato(tipoSindicato);
		jsonData['dataBase'] = dataBase != null ? DateFormat('yyyy-MM-ddT00:00:00').format(dataBase!) : null;
		jsonData['pisoSalarial'] = pisoSalarial;
		jsonData['cnpj'] = Util.removeMask(cnpj);
		jsonData['classificacaoContabilConta'] = classificacaoContabilConta;
	
		return jsonData;
	}
	
	String objectEncodeJson() {
		final jsonData = toJson;
		return json.encode(jsonData);
	}

	plutoRowToObject(PlutoRow plutoRow) {
		id = plutoRow.cells['id']?.value;
		nome = plutoRow.cells['nome']?.value;
		codigoBanco = plutoRow.cells['codigoBanco']?.value;
		codigoAgencia = plutoRow.cells['codigoAgencia']?.value;
		contaBanco = plutoRow.cells['contaBanco']?.value;
		codigoCedente = plutoRow.cells['codigoCedente']?.value;
		logradouro = plutoRow.cells['logradouro']?.value;
		numero = plutoRow.cells['numero']?.value;
		bairro = plutoRow.cells['bairro']?.value;
		municipioIbge = plutoRow.cells['municipioIbge']?.value;
		uf = plutoRow.cells['uf']?.value != '' ? plutoRow.cells['uf']?.value : 'AC';
		fone1 = plutoRow.cells['fone1']?.value;
		fone2 = plutoRow.cells['fone2']?.value;
		email = plutoRow.cells['email']?.value;
		tipoSindicato = plutoRow.cells['tipoSindicato']?.value != '' ? plutoRow.cells['tipoSindicato']?.value : 'Patronal';
		dataBase = Util.stringToDate(plutoRow.cells['dataBase']?.value);
		pisoSalarial = plutoRow.cells['pisoSalarial']?.value?.toDouble();
		cnpj = plutoRow.cells['cnpj']?.value;
		classificacaoContabilConta = plutoRow.cells['classificacaoContabilConta']?.value;
	}	

	SindicatoModel clone() {
		return SindicatoModel(
			id: id,
			nome: nome,
			codigoBanco: codigoBanco,
			codigoAgencia: codigoAgencia,
			contaBanco: contaBanco,
			codigoCedente: codigoCedente,
			logradouro: logradouro,
			numero: numero,
			bairro: bairro,
			municipioIbge: municipioIbge,
			uf: uf,
			fone1: fone1,
			fone2: fone2,
			email: email,
			tipoSindicato: tipoSindicato,
			dataBase: dataBase,
			pisoSalarial: pisoSalarial,
			cnpj: cnpj,
			classificacaoContabilConta: classificacaoContabilConta,
		);			
	}

	
}

File: src/backend/cadastros/src/model/sindicato_model.py
from src import db


class SindicatoModel(db.Model):
    __tablename__ = 'sindicato'

    id = db.Column(db.Integer, primary_key=True)
    nome = db.Column(db.String(100))
    codigo_banco = db.Column(db.Integer)
    codigo_agencia = db.Column(db.Integer)
    conta_banco = db.Column(db.String(20))
    codigo_cedente = db.Column(db.String(30))
    logradouro = db.Column(db.String(100))
    numero = db.Column(db.String(10))
    bairro = db.Column(db.String(100))
    municipio_ibge = db.Column(db.Integer)
    uf = db.Column(db.String(2))
    fone1 = db.Column(db.String(14))
    fone2 = db.Column(db.String(14))
    email = db.Column(db.String(100))
    tipo_sindicato = db.Column(db.String(1))
    data_base = db.Column(db.DateTime)
    piso_salarial = db.Column(db.Float)
    cnpj = db.Column(db.String(14))
    classificacao_contabil_conta = db.Column(db.String(30))


    def mapping(self, data):
        self.id = data.get('id')
        self.nome = data.get('nome')
        self.codigo_banco = data.get('codigoBanco')
        self.codigo_agencia = data.get('codigoAgencia')
        self.conta_banco = data.get('contaBanco')
        self.codigo_cedente = data.get('codigoCedente')
        self.logradouro = data.get('logradouro')
        self.numero = data.get('numero')
        self.bairro = data.get('bairro')
        self.municipio_ibge = data.get('municipioIbge')
        self.uf = data.get('uf')
        self.fone1 = data.get('fone1')
        self.fone2 = data.get('fone2')
        self.email = data.get('email')
        self.tipo_sindicato = data.get('tipoSindicato')
        self.data_base = data.get('dataBase')
        self.piso_salarial = data.get('pisoSalarial')
        self.cnpj = data.get('cnpj')
        self.classificacao_contabil_conta = data.get('classificacaoContabilConta')

    def serialize(self):
        return {
            'id': self.id,
            'nome': self.nome,
            'codigoBanco': self.codigo_banco,
            'codigoAgencia': self.codigo_agencia,
            'contaBanco': self.conta_banco,
            'codigoCedente': self.codigo_cedente,
            'logradouro': self.logradouro,
            'numero': self.numero,
            'bairro': self.bairro,
            'municipioIbge': self.municipio_ibge,
            'uf': self.uf,
            'fone1': self.fone1,
            'fone2': self.fone2,
            'email': self.email,
            'tipoSindicato': self.tipo_sindicato,
            'dataBase': self.data_base.isoformat(),
            'pisoSalarial': self.piso_salarial,
            'cnpj': self.cnpj,
            'classificacaoContabilConta': self.classificacao_contabil_conta,
        }